File: 백준/Silver/1940. 주몽/주몽.py
import sys
input = sys.stdin.readline

N = int(input())
M = int(input())
a = list(map(int, input().split()))
a.sort()
l = 0
r = N-1
cnt = 0
while (l < r):
    if (a[l] + a[r] < M):
        l += 1
    elif (a[l] + a[r] > M):
        r -= 1
    else:
        cnt += 1
        l += 1
        r -= 1
print(cnt)

File: 백준/Silver/1940. 주몽/주몽.java

import java.util.*;
import java.io.*;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		
		int n = Integer.parseInt(br.readLine());
		int m = Integer.parseInt(br.readLine());
		
		// 3번째 줄 입력 읽기
		StringTokenizer st = new StringTokenizer(br.readLine());
		int[] arr = new int[n];
		for (int i = 0; i < n; i++) {
			arr[i] = Integer.parseInt(st.nextToken());
		}
		Arrays.sort(arr);
		
		int s_index = 0;
		int e_index = n - 1;
		int cnt = 0;
		
		while (s_index < e_index) {
			if (arr[s_index] + arr[e_index] < m) {
				s_index++;
			} else if (arr[s_index] + arr[e_index] > m) {
				e_index--;
			} else {
				cnt++;
				s_index++;
				e_index--;
			}
			
		}
		
		System.out.println(cnt);

	}

}
